_posts/2018-01-15-trial-and-error-with-ruby-koans.markdown

```diff
@@ -0,0 +1,54 @@
+---
+layout: post
+title: "Trial and Error with Ruby Koans"
+date: 2018-01-16
+categories: ["Ruby", "Learning"]
+author: "Emily Cohen"
+---
+
+After reading introductory primers and watching instructional videos for beginners learning Ruby, I needed a program that would allow me to test out Ruby concepts for myself. I began practicing [Ruby Koans](http://rubykoans.com), **a testing-based program that runs through a variety of Ruby concepts in a trial and error fashion**.
+
+<!--more-->
+
+Instead of providing problems and requiring that the user figure out the answers, the Koans provide answers to various tests and, using *assert_equal*, require the user to fill in the blanks and correct the code until the program runs without errors.
+
+The first exercises run through Ruby concepts including **[strings](https://github.com/ercohen14/ruby-koans/blob/master/about_strings.rb)**, **[arrays](https://github.com/ercohen14/ruby-koans/blob/master/about_arrays.rb)** and **[methods](https://github.com/ercohen14/ruby-koans/blob/master/about_methods.rb)**.
+
+The exercise [About Objects](https://github.com/ercohen14/ruby-koans/blob/master/about_objects.rb) was particularly useful for me as a Junior Developer. I had heard time and again that *"in Ruby, everything is an object”*, however I didn’t understand what *“being an object”* actually entailed. This exercise went through what it means to be an object, specifying that although many items are objects, no two objects are ever exactly the same.
+
+    def test_clone_creates_a_different_object
+      obj = Object.new
+      copy = obj.clone
+
+      assert_equal true, obj           != copy
+      assert_equal true, obj.object_id != copy.object_id
+      end
+
+After the Ruby concept exercises, the Koans culminate in a series of projects that allow the user to apply what they have learned about each concept to complete more complex exercises. One that served useful to me was the [About Triangle Project](https://github.com/ercohen14/ruby-koans/blob/master/about_triangle_project.rb).
```

Comment:
@ercohen14 just wondering, wouldn't `served useful to me` read more naturally as `proved useful to me`?